gemini/sample-apps/fashion-trends-prediction/app/utils_standalone_image_gen.py

```diff
@@ -0,0 +1,605 @@
+"""
+Utility module to:
+ - Resize image bytes
+ - Generate an image with Imagen
+ - Edit an image with Imagen
+ - Render the image generation and editing UI
+"""
+
+import base64
+import io
+import math
+from typing import List
+
+import streamlit as st
+import utils_edit_image
+from config import config
+from google.cloud import aiplatform
+from google.protobuf import json_format
+from google.protobuf.struct_pb2 import Value
+from PIL import Image
+
+# Set project parameters
+PROJECT_ID = config["PROJECT_ID"]
+LOCATION = config["LOCATION"]
+
+# Set project parameters
+IMAGE_MODEL_NAME = "imagegeneration@005"
+IMAGE_MODEL_NAME_ = "imagegeneration@002"
+IMAGEN_API_ENDPOINT = f"{LOCATION}-aiplatform.googleapis.com"
+IMAGEN_ENDPOINT = f"projects/{PROJECT_ID}/locations/{LOCATION}/publishers/google/models/{IMAGE_MODEL_NAME}"
+IMAGEN_ENDPOINT_ = f"projects/{PROJECT_ID}/locations/{LOCATION}/publishers/google/models/{IMAGE_MODEL_NAME_}"
+IMAGE_UPLOAD_BYTES_LIMIT = 4096
+# The AI Platform services require regional API endpoints.
+client_options = {"api_endpoint": IMAGEN_API_ENDPOINT}
+# Initialize client that will be used to create and send requests.
+imagen_client = aiplatform.gapic.PredictionServiceClient(client_options=client_options)
```

Comment:
Why are you using the aiplatform gapic client? This can be done with the `vertexai.vision_models` module.

Reply:
This was used from the GenAI for Marketing demo code.

Reply:
Ok, please adjust your code to use the `vertexai.vision_models` module instead of the gapic client. It's recommended for most users. Likely the GenAI for Marketing demo was written before the SDK was updated or they didn't know.

Reply:
Most of these helper functions shouldn't be needed once you use the SDK functions.

Reply:
Updated - using vertexai.vision_models module now instead of the gapic client